tools/cibw_before_all.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+
+# Build and install (locally) liblsl
+
+set -eo pipefail
+ROOT=$(dirname $(dirname $(realpath "$0")))
+cd $ROOT
+echo "Using project root \"${ROOT}\" on RUNNER_OS=\"${RUNNER_OS}\""
+
+if [[ "$RUNNER_OS" == 'Linux' ]]; then
+    echo "Compiling on Linux"
+elif [[ "$RUNNER_OS" == 'macOS' ]]; then
+    echo "Compiling on macOS"
+elif [[ "$RUNNER_OS" == 'Windows' ]]; then
+    echo "Compiling on Windows"
+else
+    echo "Unknown RUNNER_OS: ${RUNNER_OS}"
+    exit 1
+fi
+
+cd src/liblsl
+# https://labstreaminglayer.readthedocs.io/dev/lib_dev.html#configuring-the-liblsl-project
+set -x
+if [[ "$RUNNER_OS" == 'macOS' ]]; then
+    cmake -S . -B build \
+        -DCMAKE_BUILD_TYPE=Release \
+        -DCMAKE_OSX_DEPLOYMENT_TARGET=10.12 \
+else
+    cmake -S . -B build -DCMAKE_BUILD_TYPE=Release
+fi
```

Comment:
So I moved the folders around to get the structure:

```
src
|_ mne_lsl
|_ liblsl
tests
```


And I don't know why, but it won't build anymore if I don't specify `MAKE_OSX_DEPLOYMENT_TARGET=10.12`. @larsoner If you have any idea as to what might be the reason.. :exploding_head: 

For now, I'm trying to fix the build as it is configured now with the new paths; and then I will try to move the build process to `setup.py`.

Reply:
That's weird!

Also be careful because the min target is 11 on arm64 IIRC, probably need to triage based on the arch